tests/end_to_end/models/participants.py

```diff
@@ -0,0 +1,431 @@
+    ModelOwner class to handle the model related operations.
+    Note: Aggregator can also act as a model owner.
+    This includes (non-exhaustive list):
+    1. Creating the workspace - to create a workspace using given workspace and model names.
+    2. Modifying based on input params provided and initializing the plan.
+    3. Certifying the workspace and setting up the PKI.
+    4. Importing and exporting the workspace etc.
+    """
+
+    def __init__(self, workspace_name, model_name):
+        """
+        Initialize the ModelOwner class
+        Args:
+            workspace_name (str): Workspace name
+            model_name (str): Model name
+        """
+        self.workspace_name = workspace_name
+        self.model_name = model_name
+        self.aggregator = None
+        self.collaborators = []
+        self.workspace_path = None
+        self.plan_path = None
+        self.num_collaborators = constants.NUM_COLLABORATORS
+        self.rounds_to_train = constants.NUM_ROUNDS
+
+    def create_workspace(self, results_dir=None):
+        """
+        Create the workspace for the model
+        Args:
+            results_dir (str): Results directory path
+        Returns:
+            str: Path to the workspace
+        """
+        try:
+            results_dir = results_dir if results_dir else os.getcwd()
+            return_code, _, error = sh.run_command(
+                f"fx workspace create --prefix {self.workspace_name} --template {self.model_name}",
+                work_dir=results_dir,
+            )
+            if return_code != 0:
+                log.error(f"Failed to create the workspace: {error}")
+                raise Exception(f"Failed to create the workspace: {error}")
+
+            log.info(f"Created the workspace {self.workspace_name} for the {self.model_name} model")
+            self.workspace_path = os.path.join(results_dir, self.workspace_name)
+            log.info(f"Workspace path: {self.workspace_path}")
+        except Exception as e:
+            log.error(f"Failed to create the workspace: {e}")
+            raise e
+        return self.workspace_path
+
+    def get_workspace_path(self, results_dir, workspace_name):
+        """
+        Get the workspace path
+        Args:
+            results_dir (str): Results directory path
+            workspace_name (str): Workspace name
+        Returns:
+            str: Path to the workspace
+        """
+        workspace_path = os.path.join(results_dir, workspace_name)
+        log.info(f"Workspace path: {workspace_path}")
+        if os.path.exists(workspace_path):
+            self.workspace_path = workspace_path
+            log.info(f"Workspace path: {self.workspace_path}")
+        else:
+            log.error(f"Workspace {workspace_name} does not exist in {results_dir}")
+            raise FileNotFoundError(f"Workspace {workspace_name} does not exist in {results_dir}")
+        return self.workspace_path
+
+    def setup_pki(self, collaborator_name):
```

Comment:
My apologies for the confusion, I back-track on my suggestion for renaming, as this method only signs the collaborator's CSR. Let's rename to `certify_collaborator()`, for symmetry with `certify_aggregator()` suggested below.

Reply:
Done. Renamed to `certify_collaborator` and `certify_aggregator`